### ISSUE TITLE
Iteration Plan January 2020

### ISSUE THREAD

**Issue by samanyougarg** [gita/BhagavadGita#394]
This plan captures our work in **January**.  This iteration will mainly be focused on getting contributors to the project, writing down the requirements for the features and planning the overall tech architecture of the project. Everyone is encouraged to contribute.

Please refer to the [Roadmap](https://github.com/gita/BhagavadGita/wiki/Roadmap) before starting.

### 1. Verse of the day
👉Every day a new verse is selected and sent to users via email, push notifications etc. Similar to how it is implemented in v1. 
👉Could rethink the process of selecting verses.

### 2. Full-Text Search
👉Should be fast and in realtime.
👉Could use something like Algolia.
👉Search by chapter, verse, phrase or person to quickly find what you are looking for.
👉Language independent.

### 3. Verse Page
- Each verse should have - >
👉Sanskrit verse
👉Transliteration of the Sanskrit verse
👉 Word meanings
👉Translation by an authorized source
👉Commentary by an authorized source
👉Audio (See feature 4 for more detail)

Actions for each verse page / on selecting a verse on the chapter page -
👉copy text
👉share text (Verse sharing should have chapter and Verse number and link for that Verse.)
👉bookmark/favourite
👉add notes (See feature 18)
👉highlight

### 4. Audio for verses
👉Audio for the Sanskrit verse, translation as well as commentary for all the verses.
👉 Initial Hindi and English support and later add for more languages.
👉Use Alexa or Google Wavenet for text to speech.

### 5. Progress Tracking  
👉User should be able to keep track of their Chapter-wise, Overall and Monthly Progress using detailed graphs and statistics. (something to similar to what's there in v1)
👉Add a GitHub like streak chart.
👉Time connected with Krishna (daily/weekly/monthly)

### 6. Achievements System
👉 Reward user with badges etc on completing certain tasks (eg 100 verses read). 
👉 Allow user to share the achievement.

### 7. Reading Plans  
👉 Allow users to subscribe to reading plans so that they can improve their Gita reading habits.
👉 Eg - monthly, yearly etc.

### 8. Favourites 
👉 Allow users to mark verses as a favourite so that they can read those later.

### 9. Theme Customization
👉Allow users to personalize the theme by allowing them to change the - 
1. font size
2. colours

👉Add option for dark mode.

### 10. Offline availability
👉Allow users to download the verses so they can read when there is no internet connection.

### 11.  Settings
👉Choose verse transliteration language
👉Choose translation/commentary source and language
👉Choose secondary commentary source and language
👉Choose font size, style and theme

### 12. Last read verse
👉Show user's last read verse on the homepage so that they can quickly navigate to it.

### 13. Random verse
👉 Display a random verse on the homepage
👉 New verse on each refresh

### 14. Navigate to specific chapter and verse 
👉 Allow the user to go to a specific chapter and verse 

### 15. Subscription popup (full screen for mobile) 
👉 Display the first time and if not subscribed then again after x number of times.
👉 Option to subscribe later (display after some time)

### 16. Chat and forum
👉A community to chat and discuss topics related to the Bhagavad Gita.

### 17. Daily Alarm
👉Remind the user to read the Gita each day.

### 18. Notes for each verse
👉A popup that allows a user to enter notes for a particular verse.
👉Options like save, delete, share and copy.

### 19. History
👉List of all the verses read by the user in reverse chronological order.

### 20. Quiz
👉Daily quizzes on the Bhagavad Gita.

### 21. Dictionary
👉A dictionary of all the word meanings from the Gita.

### 22. Hierarchical ordering
👉Structure the shlokas methodologically. Arrange them into Chapter No. And Shloka No. and add meta accordingly in frontend as well as API. 3.67 would mean Shloka no. 67 of Chapter 3.

### 23. Setup Patreon + merchandises
👉Setup patreon for donations and merchandise store.

---

### Please feel free to contribute to the features above.

---

## Tech Architecture
**Backend**: Flask
**Frontend**: ReactJS (SSR, maybe with NextJS)
**Mobile Apps(Android and iOS)**: React Native / (Java and Swift) depending on the experience of devs.
**Database**: PostgreSQL
**CI/CD**: Travis
**Analytics**: Google Analytics
**Application Performance Monitoring**: AppDynamics
**CDN**: Cloudflare
**Unit Testing**: JEST (for JS), unittest (for Python)
**Integration Testing**: Cyprus (for JS)

**Comment by ramprax** [comment]
Kudos to you efforts!

What is the source used for the current translation & commentary?

If not already done, you may want to look at below links
https://www.gitasupersite.iitk.ac.in/
https://www.gitasupersite.iitk.ac.in/srimad?language=dv&field_chapter_value=1&field_nsutra_value=1

Several translations and commentaries in Hindi, English and Sanskrit are available there.
Also, some audio is available.

Ram

**Comment by samanyougarg** [comment]
@ramprax 
Thank you much for your valuable feedback.

Currently, we are using Swami Sivananda's translation for the English version and Jaidayal Goindka for the Hindi version. 

Yes, I have had a look at Gita Supersite. They have a really nice compilation of translations and commentaries. I don't think we can use the content available on their site as they have taken permissions from a variety of different sources. So, will need to contact individual organizations who hold the copyrights.

Would you like to become a part of the initiative? If you are interested, please join our Slack channel  - https://join.slack.com/t/thegitainitiative/shared_invite/enQtODc4ODUwNDQxNDczLTM1OTA4Y2NiNWEyM2Q0YzhhMWFkNjM5Yzk1NzhhZTI0NDVhMzI4ZjFlYzE4YWM1MWRlMTBiMmI0NDRjYjhkM2U

Thank you very much.

**Comment by agupta1989** [comment]
@samanyougarg Hi, I have experience in React and some intermediate exposure on Flask Framework (Python).

**Comment by samanyougarg** [comment]
@agupta1989 Thanks a lot for your interest. Please join our slack channel using https://join.slack.com/t/thegitainitiative/shared_invite/enQtODc4ODUwNDQxNDczLTM1OTA4Y2NiNWEyM2Q0YzhhMWFkNjM5Yzk1NzhhZTI0NDVhMzI4ZjFlYzE4YWM1MWRlMTBiMmI0NDRjYjhkM2U

**Comment by agupta1989** [comment]
> 
> 
> @agupta1989 Thanks a lot for your interest. Please join our slack channel using https://join.slack.com/t/thegitainitiative/shared_invite/enQtODc4ODUwNDQxNDczLTM1OTA4Y2NiNWEyM2Q0YzhhMWFkNjM5Yzk1NzhhZTI0NDVhMzI4ZjFlYzE4YWM1MWRlMTBiMmI0NDRjYjhkM2U

It expired. Can you please re-share it again ?

**Comment by samanyougarg** [comment]
@agupta1989 here's a new link - https://join.slack.com/t/thegitainitiative/shared_invite/enQtOTMwNTA2OTczMDc2LWYxMjNjM2RmZDU0NDk5ZjgzYTNkNjBkZTQ2YTY5YWQ2ZTY1MDFjNWMxNzFjYjZkNmE0ZmM4ZWE5ZmU1YTA0ODQ

**Comment by jayvasantjv** [comment]
> @agupta1989 here's a new link - https://join.slack.com/t/thegitainitiative/shared_invite/enQtOTMwNTA2OTczMDc2LWYxMjNjM2RmZDU0NDk5ZjgzYTNkNjBkZTQ2YTY5YWQ2ZTY1MDFjNWMxNzFjYjZkNmE0ZmM4ZWE5ZmU1YTA0ODQ

Expired again!

**Comment by samanyougarg** [comment]
@jayvasantjv Here's the new link - https://join.slack.com/t/thegitainitiative/shared_invite/zt-dclsan2f-gL2s3oj1P3UQsc5v2fKpDQ
